### DISCUSSION TITLE
Using a Redis service rather than the built in Redis module

### DISCUSSION THREAD

**Discussion by hammoaj** [Budibase/budibase#1991] [Help]
Hi - I'm trying to implement Budibase in a cloud hosted Kubernetes environment, using the cloud providers own native Redis-as-a-service (as well as their CouchDB-as-a-service and their own S3 compatible object store).

I'm coming up with issues around connecting to the Redis service as it requires TLS and looking through the code, it seems as if the Redis connection management is still quite rudimentary. Am I missing anything here? Is there a way to connect to an external Redis service (rediss://....) without some re-coding in the Redis module?

Thanks

**Comment by shogunpurple** [comment]
@hammoaj you should be able to set the `REDIS_URL`on the app server pods to point to another redis instance.

We are also currently working on a helm chart and full K8S setup for budibase, which will be available over the coming weeks!

**Reply by hammoaj** [reply]
Unfortunately it doesn't work. I set the `REDIS_URL` to `rediss://<url>:<port>` to force it to use TLS.  In `packages/auth/src/redis/utils.js` on line 17 it splits the URL line into a host and port using the : symbol. This won't work on the above URL as it splits on the first colon not the second one.

Might be better to split this using the URL module.

**Reply by shogunpurple** [reply]
@hammoaj thanks for that - looks like we don't account for the prefix in the redis URL. I'll create an issue from this discussion and address it!

**Reply by vicam001** [reply]
Hi @hammoaj , I am also interested on achieving this: on a self-hosted scenario, running the docker-compose.yaml setup on an AWS EC2 instance , instead of running a Redis container I wold like to connect to AWS ElastiCache for Redis.

I have found a related discussion #2895 , that might be of your interest.

Please let me know if you have finally figured out how to achieve this, thanks!
